Tools/debug/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,248 @@
+#  event logs, driver lists, registry information, update logs,
+#  services, uptime, running processes, installed applications,
+#  installed KBs, and memory dumps.
+
+#  Copyright (c) 2024 DAYNIX LTD. All rights reserved.
+
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions
+#  are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the names of the copyright holders nor the names of their contributors
+#     may be used to endorse or promote products derived from this software
+#     without specific prior written permission.
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+#  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE
+#  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+#  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+#  HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+#  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+#  OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+#  SUCH DAMAGE.
+                       
+
+param (
+    [switch]$IncludeSensitiveData
+)
+
+function Get-SystemConfiguration {
+    try {
+        Start-Process -FilePath "msinfo32.exe" -ArgumentList "/report", "msinfo32.txt" -Wait
+    } catch {
+        Write-Warning "Failed to collect system configuration: $_"
+    }
+}
+
+function Get-EventLogs {
+    try {
+        $logTypes = @("system", "security", "application")
+        foreach ($logType in $logTypes) {
+            wevtutil epl $logType "$logType.evtx"
+            wevtutil al "$logType.evtx"
+        }
+    } catch {
+        Write-Warning "Failed to collect event logs: $_"
+    }
+}
+
+function Get-DriversList {
+    if (-not ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
+        Write-Warning "This function requires administrative privileges. Please run the script as an administrator."
+        return
+    }
+    try {
+        $drivers = Get-WindowsDriver -Online -All
+        if ($drivers) {
+            $drivers | Select-Object -Property * | Export-Csv -Path drv_list.csv -NoTypeInformation
+        } else {
+            Write-Warning "No drivers found."
```

Comment:
I wonder what happens if you omit the check. I suspect it is unnecessary. You don't have such a check for other `Export-Csv` calls in fact.